src/if_statements.py

```diff
@@ -762,29 +738,34 @@ def build_body(context: Context, function_info: FunctionInfo, options: Options)
                 if has_common and target == most_common and index != 0:
                     continue
                 context.case_nodes[target].append((switch_index, index))
-        elif isinstance(node, ConditionalNode) and node.is_loop():
-            context.loop_nodes.add(node.conditional_edge)
-        elif isinstance(node, BasicNode) and node.is_loop():
-            context.loop_nodes.add(node.successor)
 
     if options.debug:
         print("Here's the whole function!\n")
+
     body: Body
-    if options.ifs:
-        body = build_flowgraph_between(context, start_node, return_node, 1)
+    if options.ifs and is_reducible:
+        body = build_flowgraph_between(context, start_node, terminal_node)
+        body.elide_empty_returns()
     else:
-        body = build_naive(context, context.flow_graph.nodes, return_node)
-
-    if return_node.index != -1:
-        add_return_statement(context, body, return_node, 1, last=True)
+        body = build_naive(context, context.flow_graph.nodes)
+
+    # Check no nodes were skipped: build_flowgraph_between should hit every node in
+    # well-formed (reducible) graphs; and build_naive explicitly emits every node
+    unemitted_nodes = (
+        set(context.flow_graph.nodes)
+        - context.emitted_nodes
+        - {context.flow_graph.terminal_node()}
+    )
+    for node in unemitted_nodes:
+        body.add_comment(f"bug: did not emit code for node #{node.name()}")
```

Comment:
I wonder if we should expand this to also include the contents of said block?

Reply:
Sure - not too hard.